apps/vscode/src/main.ts

```diff
@@ -259,3 +269,37 @@ function registerWorkspaceFileWatcher(
 
   context.subscriptions.push(workspaceFileWatcher);
 }
+
+let codeLensProvider: Disposable | null;
+function registerWorkspaceCodeLensProvider(context: ExtensionContext) {
+  if (GlobalConfigurationStore.instance.get('enableWorkspaceConfigCodeLens')) {
+    codeLensProvider = languages.registerCodeLensProvider(
+      { pattern: '**/{workspace,angular}.json' },
+      new WorkspaceCodeLensProvider()
+    );
+    context.subscriptions.push(codeLensProvider);
+  }
+}
+
+function watchWorkspaceCodeLensConfigChange(context: ExtensionContext) {
+  context.subscriptions.push(
+    workspace.onDidChangeConfiguration((event: ConfigurationChangeEvent) => {
+      // if the `nxConsole` config changes, check enableWorkspaceConfigCodeLens and register or dispose
+      if (
+        event.affectsConfiguration(
+          GlobalConfigurationStore.configurationSection
+        )
+      ) {
+        const enableWorkspaceConfigCodeLens = GlobalConfigurationStore.instance.get(
+          'enableWorkspaceConfigCodeLens'
+        );
+        if (enableWorkspaceConfigCodeLens && !codeLensProvider) {
+          registerWorkspaceCodeLensProvider(context);
+        } else if (!enableWorkspaceConfigCodeLens && codeLensProvider) {
+          codeLensProvider.dispose();
+          codeLensProvider = null;
+        }
+      }
+    })
+  );
```

Comment:
Nice! This looks like it works properly :smile:

Do you think this logic should be handled in the workspace lib? Like have a main entry to that lib, and then do all this logic in there?

Reply:
That makes sense. Can't keep throwing everything in main/activate.